### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by alexdor**
Updated to newer packages in order to leverage new typescript features, also it fixes some vulnerabilities that npm was reporting

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,26 +35,26 @@
...
     "json-rpc2": "^1.0.2",
-    "vscode-debug-logger": "^0.0.4",
-    "vscode-debugadapter": "^1.30.0",
-    "vscode-debugprotocol": "^1.11.0",
+    "vscode-debug-logger": "0.0.5",
+    "vscode-debugadapter": "^1.32.1",
+    "vscode-debugprotocol": "^1.32.0",
     "vscode-extension-telemetry": "^0.0.22",
-    "vscode-languageclient": "~4.3.0"
+    "vscode-languageclient": "^5.1.0"
```
Comment (ramya-rao-a) [inline package.json]: The last time I changed the major version of the language client, it broke the language server feature :)
Did you get a chance to test the language server?

File: package.json [hunk 2]
```diff
@@ -35,26 +35,26 @@
...
-    "tslint": "^4.0.2",
-    "typescript": "^2.1.5",
+    "fs-extra": "^7.0.0",
+    "tslint": "^5.11.0",
+    "typescript": "^3.1.3",
     "vscode": "^1.1.21"
   },
   "engines": {
-    "vscode": "^1.22.0"
+    "vscode": "^1.26.0"
```
Comment (ramya-rao-a) [inline package.json]: Any particular reason for this change?
This change will result in the next update to the Go extension not being available to anyone who is on versions of VS Code older than 1.26.0

File: test/go.test.ts [hunk 3]
```diff
@@ -804,7 +801,7 @@ It returns the number of bytes written and any write error encountered.
 			return vscode.window.showTextDocument(textDocument).then(editor => {
 				let promises = testCases.map(([position, expected]) =>
 					provider.provideCompletionItems(editor.document, position, null).then(items => {
-						let labels = items.map(x => x.label);
+						let labels = items.items.map(x => x.label);
```
Comment (ramya-rao-a) [inline test/go.test.ts]: I merged another PR that fixed the tests, can you revert your changes to this file?

File: package.json (hunk 2 above)
Comment (alexdor) [reply package.json]: There where typing issues with one of the dependencies with the 1.22 version, I based my solution on this https://github.com/Microsoft/language-server-protocol/issues/556#issuecomment-416884011 comment

Comment (ramya-rao-a) [reply package.json]: In that case, lets not update any of the `vscode-` dependencies just yet.

I usually prefer to wait till we are atleast 3 versions away from the minimum specified in the `engine` property. Unless, of course there is a bug fix that we need to take in immediately.

**Comment by ramya-rao-a** [comment]
Thanks for the PR @alexdor!

Also, I see that this is your first contribution to vscode-go. Welcome & Thanks!
I hope that you have registered both for the [Microsoft Hacktoberfest](https://open.microsoft.com/2018/09/18/hacktoberfest-2018-microsoft/) and the one from [Digitalocean](https://hacktoberfest.digitalocean.com/)

Happy Coding!

**Pull request merged**